modules/key-card/src/generateQrData.ts

```diff
@@ -0,0 +1,171 @@
+  // If both the passphrase and passcodeEncryptionCode are passed, then this code encrypts the passphrase with the
+  // passcodeEncryptionCode and puts the result into Box D. Allows recoveries of the wallet password.
+  passphrase?: string;
+  // The user keychain as it is returned from the BitGo API upon creation
+  userKeychain: Keychain;
+  // The key id of the user key, only used for cold keys
+  userMasterKey?: string;
+}
+
+interface QrDataEntry {
+  data: string;
+  description: string;
+  image: string;
+  title: string;
+  publicMasterKey?: string;
+}
+
+export interface QrData {
+  backup?: QrDataEntry;
+  bitgo?: QrDataEntry;
+  passcode?: QrDataEntry;
+  user: QrDataEntry;
+}
+
+function getPubFromKey(key: Keychain): string | undefined {
+  if (key.type === 'blsdkg' && key.commonPub) {
+    return key.commonPub;
+  }
+  if (key.type === 'tss' && key.commonKeychain) {
+    return key.commonKeychain;
+  }
+  if (key.type === 'independent' && key.pub) {
+    return key.pub;
+  }
+}
+
+function generateUserQrData(userKeychain: Keychain, userMasterKey?: string): QrDataEntry {
+  if (userKeychain.encryptedPrv) {
+    return {
+      title: 'A: User Key',
+      image: '.qrEncryptedUserKey',
+      description: 'This is your private key, encrypted with your wallet password.',
+      data: userKeychain.encryptedPrv,
+    };
+  }
+
+  const pub = getPubFromKey(userKeychain);
+  assert(pub);
+
+  return {
+    title: 'A: Provided User Key',
+    image: '.qrPublicUserKey',
+    description: 'This is the public key you provided for your wallet.',
+    data: pub,
+    publicMasterKey: userMasterKey,
+  };
+}
+
+function generateBackupQrData(coin: Readonly<BaseCoin>, backupKeychain: Keychain, {
+  backupKeyProvider,
+  backupMasterKey,
+}: {
+  backupKeyProvider?: string;
+  backupMasterKey?: string;
+} = {}): QrDataEntry {
+  const title = 'B: Backup Key';
+  if (backupKeychain.encryptedPrv) {
+    return {
+      title,
+      image: '.qrEncryptedBackupKey',
+      description: 'This is your backup private key, encrypted with your wallet password.',
+      data: backupKeychain.encryptedPrv,
+    };
+  }
+
+  const pub = getPubFromKey(backupKeychain);
+  assert(pub);
+
+  if (backupKeyProvider) {
+    return {
+      title: 'B: Backup Key',
+      image: '.qrPublicBackupKey',
+      description:
+        'This is the public key held at ' +
+        backupKeyProvider +
+        ', an ' +
+        coin.name +
+        ' recovery service. If you lose\r\nyour key, ' +
+        backupKeyProvider +
+        ' will be able to sign transactions to recover funds.',
+      data: pub,
+    };
+  }
+
+  return {
+    title: 'B: Provided Backup Key',
+    image: '.qrEncryptedUserProvidedXpub',
+    description: 'This is the public key you provided for your wallet.',
+    data: pub,
+    publicMasterKey: backupMasterKey,
+  };
+}
+
+function generateBitGoQrData(coin: Readonly<BaseCoin>, bitgoKeychain: Keychain): QrDataEntry | undefined {
```

Comment:
this would never return `undefined` ?

Reply:
also not sure where `coin` is used here.

Reply:
Good catch. Both were leftover from when I was supporting the OFC special case. Removing them now.